### PR TITLE
task 5 done

### PR DESCRIPTION
**Pull request by crypticalfish86**
task 5 done

### CODE REVIEW

File: app/model.js [hunk 1]
```diff
@@ -11,4 +11,33 @@ fetchTopics = () =>
...
+        {
+            return db.query
+            (
+                `SELECT *
+                FROM articles
+                WHERE article_id = ${params}`
+            )
+            .then((response) =>
+            {
+                if(response.rowCount !== 0)
```
Comment (MrMcSnail) [inline app/model.js:56]: Nice use of .rowCount

File: __tests__/app.test.js [hunk 2]
```diff
@@ -50,4 +50,74 @@ describe('APP be-nc-news', () =>
...
     })
-})
\ No newline at end of file
+    describe('GET /api/articles/:article_id', () =>
+    {
+        test('GET: will return with a status of 200', () =>
+        {
+            return request(app).get('/api/articles/1').expect(200)
+        })
+        test('responds with an object that isn\'t empty', () =>
```
Comment (MrMcSnail) [inline __tests__/app.test.js]: nice incremental TDD. But maybe too incremental! 
This test is made redundant by the next one.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -50,4 +50,74 @@ describe('APP be-nc-news', () =>
...
+                expect(Object.keys(body).length !== 0).toBe(true)
+            })
+        })
+        test(`response is an object with the object keys: author, title
+        article_id, body, topic, created_at, votes, article_img_url`, () =>
+        {
+            return request(app).get('/api/articles/1').expect(200)
+            .then(({ body }) =>
+            {
+                expect(body).toHaveProperty('author')
```
Comment (MrMcSnail) [inline __tests__/app.test.js]: Have a look  at (`.objectContaining`)[https://jestjs.io/docs/expect#expectobjectcontainingobject], you might prefer in circumstances like this, but its just a matter of preferance.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -50,4 +50,74 @@ describe('APP be-nc-news', () =>
...
+            })
+        })
+        test(`response is an object with the object keys: author, title
+        article_id, body, topic, created_at, votes, article_img_url
+        where the relevent article_id is present`, () =>
+        {
+            return request(app).get('/api/articles/1').expect(200)
+            .then(({ body }) =>
+            {
+                expect(body).toHaveProperty('author')
```
Comment (MrMcSnail) [inline __tests__/app.test.js]: as above

File: __tests__/app.test.js [hunk 5]
```diff
@@ -50,4 +50,74 @@ describe('APP be-nc-news', () =>
...
+        })
+        test('responds with error 400 if anything except a number is request as a parameter', () =>
+        {
+            return request(app).get('/api/articles/test').expect(400)
+            .then((err) =>
+            {
+                expect(err.body.msg).toEqual('invalid article ID: not a number')
+            })
+        })
+        test('responds with error 400 if URL specifies an ID that does not exist', () =>
```
Comment (MrMcSnail) [inline __tests__/app.test.js]: Good work!

File: __tests__/app.test.js [hunk 6]
```diff
@@ -50,4 +50,74 @@ describe('APP be-nc-news', () =>
...
+        {
+            return request(app).get('/api/articles/132958323').expect(400)
+            .then((err) =>
+            {
+                expect(err.body.msg).toEqual('invalid article ID: ID not found')
+            })
+        })
+    })
+})
+/*if you are using send() in anyway you are sending a body which will then
```
Comment (MrMcSnail) [inline __tests__/app.test.js]: make sure these comments are left out of commits

**Review by MrMcSnail — APPROVED** [review]
Great job. Merge away!!

File: __tests__/app.test.js [hunk 7]
```diff
@@ -50,4 +50,74 @@ describe('APP be-nc-news', () =>
...
+        {
+            return request(app).get('/api/articles/test').expect(400)
+            .then((err) =>
+            {
+                expect(err.body.msg).toEqual('invalid article ID: not a number')
+            })
+        })
+        test('responds with error 400 if URL specifies an ID that does not exist', () =>
+        {
+            return request(app).get('/api/articles/132958323').expect(400)
```
Comment (Hard-Luck) [inline __tests__/app.test.js]: this should be a 404, it will need changing in the model and the test file

**Review by Hard-Luck — COMMENTED** [review]
Just the one change then you can merge

**Pull request merged**